src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -533,24 +535,41 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         return _connection.State();
     }
 
-    void TermControl::SwapChainChanged()
+    winrt::fire_and_forget TermControl::SwapChainChanged()

```

Comment:
While reading some background documentation on how this whole co* stuff worked, I noticed that `SwapChainPanel` apparently has a provision to let background threads potentially mess with the chain instead of having to jump to the UI thread to do so:

https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.swapchainpanel.createcoreindependentinputsource#Windows_UI_Xaml_Controls_SwapChainPanel_CreateCoreIndependentInputSource_Windows_UI_Core_CoreInputDeviceTypes_

It's probably not easily/instantly applicable here. But I want to leave the note because I thought of it and @DHowett-MSFT and @zadjii-msft might have an opinion on whether that would be a decent follow-on task to generate from this comment.

Reply:
Moved to #4164 for further investigation.